chia/full_node/mempool.py

```diff
@@ -99,23 +93,14 @@ def add_to_pool(self, item: MempoolItem) -> None:
             if coin_id not in self.removals:
                 self.removals[coin_id] = []
             self.removals[coin_id].append(item.name)
-        self.total_mempool_cost += item.cost
 
-        mempool_info = self.get_mempool_info()
-        self.fee_estimator.add_mempool_item(mempool_info, item)
+        current_mempool_cost = self.mempool_info.current_mempool_cost + item.cost
+        self.mempool_info = dataclasses.replace(self.mempool_info, current_mempool_cost=current_mempool_cost)
```

Comment:
here too. should `mempool_info` just be part of the fee estimator?

Reply:
Not as-is, no. There are parts that update on every call from the mempool into the estimator. (current mempool size, for example).

Reply:
So far, I'm thinking that all I need to do is pull current_mempool_cost out - the other members are set at MempoolManager construction time. I'll try that first.

Reply:
I like your idea of letting FeeMempoolInfo live in the fee estimator - I think that would be better overall.
As you can see, these are mostly constant throughout the life of the FullNode.

A couple downsides are:
* Data is different places, possibly duplicate
* No ability to dynamically set some of these values (like max mempool size) without restart

I'm not tied to point number 2 - the scenario I thought it would be useful in is to have a full-system test on a testnet where all nodes set very low blocksize and mempool sizes.

My experience with this project is that most of the bugs have been from integrating the fee estimator with the code that calls it - it is much harder to test multiple connected full nodes, or even a single full node (although that is becoming easier with `FullNodeSimulator`).